src/main/java/org/asamk/signal/json/JsonContactEmail.java

```diff
@@ -0,0 +1,30 @@
+package org.asamk.signal.json;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import org.whispersystems.signalservice.api.messages.shared.SharedContact;
+
+public class JsonContactEmail {
+
+    @JsonProperty
+    private final String value;
+
+    @JsonProperty
+    private final SharedContact.Email.Type type;
+
+    @JsonProperty
+    private final String label;
+
+    private String getValueIfPresent(String string) {
```

Comment:
Maybe change this method to take an `Optional<String>`?
Also you can move it as a static method to `util/Util.java` and reuse it.

Reply:
All good. For some reason I thought you didn't want Optionals in the codebase, but I'll fix that now.

Reply:
Yeah, I'm not too fond of Optionals, but for the helper method it's totally fine.